openraft/src/raft.rs

```diff
@@ -570,7 +570,11 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> Raft<C, N,
             let _ = tx.send(());
         }
         if let Some(handle) = self.inner.raft_handle.lock().await.take() {
-            let _ = handle.await?;
+            {
+                let mut guard = self.inner.core_result.lock().await;
+                let _ = handle.await?;
+                *guard = Ok(());
+            } // Atomic update core error.
```

Comment:
You need the lock `raft_handle` to protect these steps: `take(), handle.await, and setting the core_result`.

And everywhere using `raft_handle.lock().await.take()` should have the same procedure.

Reply:
Done.